apis/submariner/v1alpha1/submariner_types.go

```diff
@@ -94,6 +95,15 @@ type DaemonSetStatus struct {
 	MismatchedContainerImages bool                     `json:"mismatchedContainerImages"`
 }
 
+type DeploymentInfo struct {
+	// +kubebuilder:validation:Enum=k8s;ocp;eks;aks;gke
```

Comment:
Can we do it without the validations?

I'd leave it free form. There are more types, like microk8s, k3s, etc... and more that we don't know. If we apply too much validation then we will experience issues during upgrades I believe.

I belive it should be ok, because it's not an user-input, but instead a report that we make. We can add  constants in this file so we make sure that we will use those constants when filling up such fields

Reply:
Agree. I prefer it free form too but realized no one will have an opinion if I don't add validations.

Reply:
Let's convert them to constants that we can use when reporting, and we can add more later :)

Reply:
What about k3s?

Reply:
Yeah, exactly, that's why I was proposing to avoid validation, as it's not user-input but status that we report.